### PR TITLE
Upgrade coreos/go-oidc to v3 to pickup claims parsing fixes

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
[Reference conversation](https://flyte-org.slack.com/archives/CP2HDHKE1/p1695727130623009)

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

**Comment by eapolinario** [comment]
Hi, we are moving all Flyte development to a monorepo. In order to help the transition period, we're moving open PRs to monorepo automatically and your PR was moved to https://github.com/flyteorg/flyte/pull/4139. Notice that if there are any conflicts in the resulting PR they most likely happen due to the change in the import path of the flyte components.
